### ISSUE TITLE
Whisk.invoke object is not working in Swift3 container

### ISSUE THREAD

**Issue by paulcastro** [openwhisk/openwhisk#1705]
The Whisk.invoke object is not working correctly in the Swift 3 container.  If you try to use the object to invoke an action, the invoking action will always return:

```
"result": {
            "error": "The action did not return a dictionary."
        }

```

**Comment by paulcastro** [comment]
There appears to be two necessary changes:

1. the open whisk environment variable that has the hostname now seems to include the protocol.  We'll need to parse that
2. We should leave the dispatch group after the callback, not in the callback.  I'm not quite sure how this worked before, but its broken now.

**Comment by paulcastro** [comment]
I'll add a function that uses URLSession in case we want to move to that in the future.  The URLCredential in Linux doesn't have a complete implementation yet so we'd be limited to invoking actions on OpenWhisk installations that have valid certificates if we switch to it now.  This may be the preferred behavior but  not ideal for development environments.

**Comment by paulcastro** [comment]
Re-activating test cases for Whisk object.

**Comment by paulcastro** [comment]
Fixed, has passed tests.  Will submit PR

**Comment by paulcastro** [comment]
Merged PR #1718
